go-controller/pkg/ovn/gateway_init.go

```diff
@@ -183,8 +184,16 @@ func (oc *Controller) gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet,
 
 	externalRouterPort := types.GWRouterToExtSwitchPrefix + gatewayRouter
 
-	// Add static routes in GR with gateway router as the default next hop.
-	for _, nextHop := range l3GatewayConfig.NextHops {
+	nextHops := l3GatewayConfig.NextHops
+	if len(l3GatewayConfig.NextHops) == 0 {
+		nextHops = node.DummyNextHopIPs()
+		if err := gateway.CreateDummyGWMacBinding(oc.sbClient, nodeName); err != nil {
+			return err
+		}
+	}
```

Comment:
The asymmetry of using dummy IPs in the host always, but only if there is no default GW in the GW router kinda stands up. I guess it works with less changes required but I am not sure if overall going forward it would be easier to understand using the dummy IPs always on both ends.
So if there is no default gw, we don't care much we are pointing to nowhere as external traffic is not supposed to work anyway, right?

Reply:
I agree on your assessment - while we end up adding less code, it is harder to reason about it unless you know specifically what is happening.

Do you think moving this code to a (properly named) helper would make the reader less confused ?

Reply:
hmm not sure, I would be more than happy with a brief explanation in `DumyNextHopIPs` doc.